softlayer/data_source_softlayer_vlan.go

```diff
@@ -0,0 +1,74 @@
+package softlayer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+)
+
+func dataSourceSoftLayerVlan() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSoftLayerVlanRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"number", "router_hostname"},
+			},
+
+			"number": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+
+			"router_hostname": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
```

Comment:
How about other computed fields such as `datacenter`, `type`, `primary_subnet_size`, `softlayer_managed`, `child_resource_count`, and `subnets`? Especially, `subnets` can be used for provisioning automation.